test/e2e/antreapolicy_test.go

```diff
@@ -568,6 +568,61 @@ func testACNPAllowXBtoA(t *testing.T) {
 	executeTests(t, testCase)
 }
 
+// testACNPSourcePort tests ACNP source port filtering. The agnhost image used in E2E tests uses
+// ephemeral ports to initiate TCP connections, which should be 32768–60999 by default
+// (https://en.wikipedia.org/wiki/Ephemeral_port). This test retrieves the port range from
+// the client Pod and uses it in sourcePort and sourceEndPort of an ACNP rule to verify that
+// packets can be matched by source port.
+func testACNPSourcePort(t *testing.T) {
+	portStart, portEnd, err := k8sUtils.getTCPv4SourcePortRangeFromPod(namespaces["x"], "a")
+	failOnError(err, t)
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("acnp-source-port").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{PodSelector: map[string]string{"pod": "a"}}})
+	builder.AddIngressForSrcPort(ProtocolTCP, nil, nil, &portStart, &portEnd, nil, nil, nil, nil, nil, map[string]string{"pod": "b"}, map[string]string{"ns": namespaces["x"]},
+		nil, nil, false, nil, crdv1alpha1.RuleActionDrop, "", "", nil)
+
+	updatedACNPBuilder := &ClusterNetworkPolicySpecBuilder{}
+	updatedACNPBuilder = updatedACNPBuilder.SetName("acnp-source-port").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{PodSelector: map[string]string{"pod": "a"}}})
+	updatedACNPBuilder.AddIngressForSrcPort(ProtocolTCP, &p80, nil, &portStart, &portEnd, nil, nil, nil, nil, nil, map[string]string{"pod": "b"}, map[string]string{"ns": namespaces["x"]},
```

Comment:
Maybe change to use `port`, `endPort`, `srcPort` and `srcEndPort` all together case, like `AddIngressForSrcPort(ProtocolTCP, &p8080, &p8082, &portStart, &portEnd,...` to test conjunctions are correct for the multiplication between two ranges.
Except for the comment above, everything LGTM.